### PR TITLE
Fix sdlc broken when kserve installed using helm

### LINKED ISSUE #3891: SDLC is broken when KServe is installed with helm
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]
1. Install KServe using [quick_install.sh](https://github.com/kserve/kserve/blob/master/hack/quick_install.sh) which uses helm.
2. Run `make deploy-dev`

```yaml
 - name: manager
      state:
        waiting:
          reason: CrashLoopBackOff
          message: >-
            back-off 2m40s restarting failed container=manager
            pod=kserve-controller-manager-564985d655-47tts_kserve(b051885e-3bdf-4040-9e95-140793bbd90c)
      lastState:
        terminated:
          exitCode: 127
          reason: ContainerCannotRun
          message: >-
            failed to create task for container: failed to create shim task: OCI
            runtime create failed: runc create failed: unable to start container
            process: exec: "/manager": stat /manager: no such file or directory:
            unknown
          startedAt: '2024-08-26T09:00:15Z'
          finishedAt: '2024-08-26T09:00:15Z'
          containerID: >-
            docker://c1bf44299b930abdee2b65a13a657ddc1e9bcab5336bcd90ad21c6e4e9895b86
      ready: false
      restartCount: 5
```

**What did you expect to happen:**
The Kserve controller should be deployed successfully.

**What's the InferenceService yaml:**
[To help us debug please run `kubectl get isvc $name -n $namespace -oyaml` and paste the output]

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Istio Version:
- Knative Version:
- KServe Version:
- Kubeflow version:
- Cloud Environment:[k8s_istio/istio_dex/gcp_basic_auth/gcp_iap/aws/aws_cognito/ibm]
- Minikube/Kind version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

### PR DESCRIPTION
**Pull request by sivanantha321**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #3891

**Type of changes**
Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Feature/Issue validation/testing**:

Please describe the tests that you ran to verify your changes and relevant result summary. Provide instructions so it can be reproduced.
Please also list any relevant details for your test configuration.

- [ ] Test A
- [ ] Test B

- Logs

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Checklist**:

- [ ] Have you added unit/e2e tests that prove your fix is effective or that this feature works?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

**Release note**:

```release-note
Fix sdlc broken when kserve installed using helm
```

**Re-running failed tests**

- `/rerun-all` - rerun all failed workflows.
- `/rerun-workflow <workflow name>` - rerun a specific failed workflow. Only one workflow name can be specified. Multiple /rerun-workflow commands are allowed per comment.

### CODE REVIEW

File: config/webhook/manifests.yaml [hunk 1]
```diff
@@ -26,7 +25,6 @@ webhooks:
         resources:
           - inferenceservices
   - clientConfig:
-      caBundle: Cg==
```
Comment (yuzisun) [inline config/webhook/manifests.yaml:29]: Is the placeholder not needed anymore ?

Comment (sivanantha321) [reply config/webhook/manifests.yaml:29]: Yes. It is not used anywhere. The caBundle is automatically injected by the cert-manager. Having this placeholder produces conflict when we try to run the `make deploy-dev` second time as it is trying to replace the `ca bundle` with this dummy value.

**Review by yuzisun — APPROVED** [review]
/lgtm
/approve

**Pull request merged**